engine/crates/runtime/src/hot_cache.rs

```diff
@@ -0,0 +1,65 @@
+use std::future::Future;
+
+#[derive(strum::Display)]
+pub enum CachedDataKind {
+    PersistedQuery,
+    Operation,
+}
+
+pub trait HotCacheFactory: Send + Sync + 'static {
+    type Cache<V>: HotCache<V>
+    where
+        V: Clone + Send + Sync + 'static + serde::Serialize + serde::de::DeserializeOwned;
+
+    /// A new instance provides a convenient interface on how values are handled. Keys
+    /// still live in the same namespace and MUST be unique.
+    fn create<V>(&self, kind: CachedDataKind) -> impl Future<Output = Self::Cache<V>> + Send
+    where
+        V: Clone + Send + Sync + 'static + serde::Serialize + serde::de::DeserializeOwned;
+}
+
+/// Cache meant to store data in a really fast cache, ideally directly in-memory.
+/// It is *not* meant for response caching.
+/// It's up to the implementation to decide how to evict values to save space.
+///
+/// Contract:
+/// - values are immutable for a given key
+/// - values are serialize-able with postcard, and thus any serde format
```

Comment:
Postcard doesn't support everything serde does, so this wording seems a bit misleading.

Maybe ditch everything after the comma?

Reply:
ah yes, meant the opposite, if it works with postcard means it should work with anything since postcard is quite strict. but you're right

Reply:
ah ok yeah you're totally right